staging/vhost-device-gpu/src/protocol.rs

```diff
@@ -0,0 +1,1453 @@
+        assert_eq!(format!("{:?}", ctx_attach_resource), "CtxAttachResource");
+    }
+
+    #[test]
+    fn test_ctx_detach_resource_debug() {
+        let ctx_detach_resource = GpuCommand::CtxDetachResource(virtio_gpu_ctx_resource::default());
+        assert_eq!(format!("{:?}", ctx_detach_resource), "CtxDetachResource");
+    }
+
+    #[test]
+    fn test_resource_create_3d_debug() {
+        let resource_create_3d =
+            GpuCommand::ResourceCreate3d(virtio_gpu_resource_create_3d::default());
+        assert_eq!(format!("{:?}", resource_create_3d), "ResourceCreate3d");
+    }
+
+    #[test]
+    fn test_transfer_to_host_3d_debug() {
+        let transfer_to_host_3d =
+            GpuCommand::TransferToHost3d(virtio_gpu_transfer_host_3d::default());
+        assert_eq!(format!("{:?}", transfer_to_host_3d), "TransferToHost3d");
+    }
+
+    #[test]
+    fn test_transfer_from_host_3d_debug() {
+        let transfer_from_host_3d =
+            GpuCommand::TransferFromHost3d(virtio_gpu_transfer_host_3d::default());
+        assert_eq!(format!("{:?}", transfer_from_host_3d), "TransferFromHost3d");
+    }
+
+    #[test]
+    fn test_cmd_submit_3d_debug() {
+        let cmd_submit_3d = GpuCommand::CmdSubmit3d {
+            cmd_data: Vec::new(),
+            fence_ids: Vec::new(),
+        };
+        assert_eq!(format!("{:?}", cmd_submit_3d), "CmdSubmit3d");
+    }
+
+    #[test]
+    fn test_resource_create_blob_debug() {
+        let resource_create_blob =
+            GpuCommand::ResourceCreateBlob(virtio_gpu_resource_create_blob::default());
+        assert_eq!(format!("{:?}", resource_create_blob), "ResourceCreateBlob");
+    }
+
+    #[test]
+    fn test_resource_map_blob_debug() {
+        let resource_map_blob =
+            GpuCommand::ResourceMapBlob(virtio_gpu_resource_map_blob::default());
+        assert_eq!(format!("{:?}", resource_map_blob), "ResourceMapBlob");
+    }
+
+    #[test]
+    fn test_resource_unmap_blob_debug() {
+        let resource_unmap_blob =
+            GpuCommand::ResourceUnmapBlob(virtio_gpu_resource_unmap_blob::default());
+        assert_eq!(format!("{:?}", resource_unmap_blob), "ResourceUnmapBlob");
+    }
+
+    #[test]
+    fn test_update_cursor_debug() {
+        let update_cursor = GpuCommand::UpdateCursor(virtio_gpu_update_cursor::default());
+        assert_eq!(format!("{:?}", update_cursor), "UpdateCursor");
+    }
+
+    #[test]
+    fn test_move_cursor_debug() {
+        let move_cursor = GpuCommand::MoveCursor(virtio_gpu_update_cursor::default());
+        assert_eq!(format!("{:?}", move_cursor), "MoveCursor");
+    }
+
+    #[test]
+    fn test_resource_assign_uuid_debug() {
+        let resource_assign_uuid =
+            GpuCommand::ResourceAssignUuid(virtio_gpu_resource_assign_uuid::default());
+        assert_eq!(format!("{:?}", resource_assign_uuid), "ResourceAssignUuid");
+    }
+
+    #[test]
+    fn test_virtio_gpu_ctx_create_debug() {
```

Comment:
same here, not sure why the debug impl has to be tested, is this for coverage?

Reply:
Yep, for coverage :) It's harmless to do so until we can get `Debug` ignored from coverage.